sim_pipeline/gg_lens_pop.py

```diff
@@ -95,14 +112,13 @@ def draw_population(self, testarea=4 * np.pi):
         # Draw a population of galaxy-galaxy lenses within the area.
         for _ in range(num_lenses):
             lens = self._lens_galaxies.draw_deflector()
-            num_sources_range = max(1, int(np.rint(
-                np.random.normal(loc=num_sources_tested_mean, scale=num_sources_tested_mean / 5))))
-            # num_sources_range is the number of sources to be tested for each lens,
-            # which is a normal distribution with mean=num_sources_tested_mean and std=num_sources_tested_mean/5
-            for _ in range(num_sources_range):
-                source = self._source_galaxies.draw_galaxy()
-                gg_lens = GGLens(deflector_dict=lens, source_dict=source, cosmo=self.cosmo, test_area=testarea)
-                # Check the validity of the lens system
-                if gg_lens.validity_test():
-                    gg_lens_population.append(gg_lens)
+            test_area = draw_test_area(deflector=lens)
+            num_sources_range = self.get_num_sources_tested(testarea=test_area)
+            if num_sources_range > 0:
+                for _ in range(num_sources_range):
```

Comment:
here we have to be careful. You effectively make a set of single-plane lens systems when instead when num_sources_range > 1 we would expect a multi-source plane lens system (not implemented yet). You can keep it for now but write a TODO that this is not exactly what we would expect